CRM/Contribute/BAO/Contribution.php

```diff
@@ -398,8 +398,8 @@ public static function calculateMissingAmountParams(&$params, $contributionID) {
             'id' => $contributionID,
             'return' => ['total_amount', 'net_amount', 'fee_amount'],
           ]);
-          $totalAmount = isset($params['total_amount']) ? $params['total_amount'] : CRM_Utils_Array::value('total_amount', $contribution);
-          $feeAmount = isset($params['fee_amount']) ? $params['fee_amount'] : CRM_Utils_Array::value('fee_amount', $contribution);
+          $totalAmount = (float) (isset($params['total_amount']) ? $params['total_amount'] : CRM_Utils_Array::value('total_amount', $contribution));
```

Comment:
```suggestion
          $totalAmount = (isset($params['total_amount']) ? (float) $params['total_amount'] : 
 (float) CRM_Utils_Array::value('total_amount', $contribution));
          $feeAmount = (isset($params['fee_amount']) ? (float) $params['fee_amount'] : (float) CRM_Utils_Array::value('fee_amount', $contribution));
```
Your code change didn't seem to work for me when the quick form had no fee amount set at all i.e. a blank square but the above code seemed to work sensibly